### PR TITLE
using stateful set for data

### PR DESCRIPTION
**Pull request by OwainPerry**
this is an example of using stateful set for the data volumes, for reference

### CODE REVIEW

File: es-data.yaml [hunk 1]
```diff
@@ -56,7 +58,11 @@ spec:
...
-              medium: ""
-            name: "storage"
+  volumeClaimTemplates:
+  - metadata:
+      name: storage
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      resources:
+        requests:
+          storage: 12Gi
```
Comment (pires) [inline es-data.yaml]: Where does this volume comes from? Can you provide context and documentation on how to leverage on this, please?

Comment (andrewmclagan) [inline es-data.yaml]: Would be great to get an update on this?

Comment (OwainPerry) [inline es-data.yaml]: Hey sorry , been maxed out, so I just wanted to show how you can use a volume e.g. in my case EBS volume attached to the pod for data , I notice now my comment was a bit sparse... I spent some time trying to use persistent volumes , but the naming causes problems , unless there is a trick I don't know with stateful sets I don't know to get them to choose them (still on a learning curve with k8s)?

Comment (andrewmclagan) [inline es-data.yaml]: @OwainPerry we are actually having the exact same issue, although on GKE.

**Comment by wfhartford** [comment]
I'm using something very close to this version of the data node stateful set in GKE.

Using a `volumeClaimTemplate` with a `storage-class` annotation allows GKE to allocate a persistent disk associated with each pod in the set:
```yaml
  volumeClaimTemplates:
  - metadata:
      name: storage
      annotations:
        volume.beta.kubernetes.io/storage-class: standard
    spec:
      accessModes: [ "ReadWriteOnce" ]
      resources:
        requests:
          storage: 12Gi
```
Then adding an appropriate `StorageClass` object tells kubernetes how to create a the disk:
```yaml
kind: StorageClass
apiVersion: storage.k8s.io/v1beta1
metadata:
  name: standard
provisioner: kubernetes.io/gce-pd
parameters:
  type: pd-standard
  zone: us-west1-b
```
@pires I can submit a pull request if you like. How should I structure the files? Maybe under a GKE sub-directory?

**Comment by pires** [comment]
@wfhartford yes, a separate folder and proper documentation would work.

**Comment by pires** [comment]
Closed by #80.

**Pull request closed without merging**